opentracing-apache-httpclient/src/main/java/io/opentracing/contrib/apache/http/client/ApacheClientSpanDecorator.java

```diff
@@ -56,12 +56,20 @@
 
         @Override
         public void onRequest(HttpRequestWrapper request, HttpContext httpContext, Span span) {
+            URI uri = request.getURI();
+            HttpHost target = request.getTarget();
+
             Tags.HTTP_METHOD.set(span, request.getRequestLine().getMethod());
 
-            URI uri = request.getURI();
-            Tags.HTTP_URL.set(span, request.getRequestLine().getUri());
-            Tags.PEER_PORT.set(span, uri.getPort() == -1 ? 80 : uri.getPort());
-            Tags.PEER_HOSTNAME.set(span, uri.getHost());
+            if (uri != null) {
+                Tags.HTTP_URL.set(span, request.getRequestLine().getUri());
+                Tags.PEER_HOSTNAME.set(span, uri.getHost());
+                Tags.PEER_PORT.set(span, uri.getPort() == -1 ? uri.getScheme().equalsIgnoreCase("https") ? 443 : 80 : uri.getPort());
+            } else if (target != null) {
+                Tags.HTTP_URL.set(span, request.getTarget() + request.getRequestLine().getUri());
```

Comment:
` + request.getRequestLine().getUri()` is this correct? The same is used when uri is not `null`.

Reply:
I found that this method returns the URI path without the scheme, host, and port for a BasicHttpRequest.  For other objects (when uri is not null) this returns the full scheme, host, port, path, etc.  To keep the tag value consistent I prepended the request.getTarget() which contains the scheme, host, and port.